### PR TITLE
add --esm and --minified options for visual-diff task

### PR DESCRIPTION
**Pull request by erwinmombay**
use --esm for router serving when possible

### CODE REVIEW

File: build-system/server/new-server/router.js [hunk 1]
```diff
@@ -2,8 +2,20 @@ const router = require('express').Router();
 // @ts-ignore
 const {transform} = require('./transforms/dist/transform');
 
-router.get('/examples/*.html', async (req, res) => {
+// There are some pages that we do not want transformed explicitly.
+const exemptPages = [
```
Comment (rsimha) [inline build-system/server/new-server/router.js:6]: I fear that this list isn't easily discoverable / maintainable and may go out of date when new js-only tests are added. Would it make sense to move these pages into a subdirectory that doesn't match the regex on line 14? (Currently `/examples/*.html`.)

**Review by rsimha — APPROVED** [review]
Overall LGTM, let's get Daniel to take a look before merging.

File: build-system/server/new-server/router.js (hunk 1 above)
Comment (erwinmombay) [reply build-system/server/new-server/router.js:6]: would it be ok to go with anything that ends with `nomodule.html` instead will skip the transformations

Comment (rsimha) [reply build-system/server/new-server/router.js]: Good idea, I think that'll work!

File: build-system/tasks/visual-diff/index.js [hunk 2]
```diff
@@ -702,12 +702,14 @@ visualDiff.description = 'Run the AMP visual diff tests';
 visualDiff.flags = {
   'main': 'Include a blank snapshot (baseline for skipped builds)',
   'empty': 'Create a dummy Percy build with only a blank snapshot',
+  'esm': 'Serve ESM JS (uses the new typescript server transforms)',
```
Comment (danielrozenberg) [inline build-system/tasks/visual-diff/index.js:705]: I have a feeling this `new` thing would become quite old by the time anyone bothers to remove this parenthesis. Is there a strong reason to keep this comment in the flag description?

File: build-system/server/app.js [hunk 3]
```diff
@@ -73,10 +73,7 @@ app.use(bodyParser.text());
 // Middleware is executed in order, so this must be at the top.
 // TODO(#24333): Migrate all server URL handlers to new-server/router and
 // deprecate app.js.
-// TODO(erwinmombay, #32865): Make visual diff tests use the new server
-if (!argv._.includes('visual-diff')) {
-  app.use(require('./new-server/router'));
-}
+app.use(require('./new-server/router'));
```
Comment (danielrozenberg) [inline build-system/server/app.js:76]: Can we split the new server code into a separate PR?

File: build-system/server/new-server/router.js [hunk 4]
```diff
@@ -2,8 +2,14 @@ const router = require('express').Router();
 // @ts-ignore
 const {transform} = require('./transforms/dist/transform');
 
-router.get('/examples/*.html', async (req, res) => {
+router.get('/examples/*.html', async (req, res, next) => {
   let transformedHTML;
+  // nomodule.html will force an opt-out of the transformation. This can
+  // be useful if we want to explicitly test the nomodule version of the
+  // code.
+  if (/\.nomodule\.html$/.test(req.path)) {
```
Comment (danielrozenberg) [inline build-system/server/new-server/router.js:10]: This feels a bit hackish and difficult to discover, I wonder if there's a way to be more explicit about this. e.g., can we add an HTTP header (something like `X-AMP-No-Module`) and explicitly add this to the .jsonc file as a field? (`force-nomodule: true`)

Otherwise if we decide to stick with this method in the end:
```suggestion
  if (req.path.endsWith('.nomodule.html')) {
```

Comment (erwinmombay) [reply build-system/server/new-server/router.js]: i like the headers idea. ill implement

File: build-system/server/app.js (hunk 3 above)
Comment (erwinmombay) [reply build-system/server/app.js:76]: i could, but it would essentially be this single line change? unless im misunderstanding your request

Comment (danielrozenberg) [reply build-system/server/app.js:76]: I mean, also update the server test files in that separate PR

File: build-system/tasks/visual-diff/index.js (hunk 2 above)
Comment (erwinmombay) [reply build-system/tasks/visual-diff/index.js:705]: no reason. will remove

File: build-system/server/app.js (hunk 3 above)
Comment (erwinmombay) [reply build-system/server/app.js:76]: isolated the server changes in https://github.com/ampproject/amphtml/pull/37121/files (that needs to go in first) then this one for the visual diff changes

**Comment by erwinmombay** [comment]
@danielrozenberg PTAL

**Review by danielrozenberg — APPROVED** [review]
